### ISSUE TITLE
Argument names for waveforms, 2d representations (real and complex)

### ISSUE THREAD

**Issue by keunwoochoi** [keunwoochoi/torchaudio-contrib#46]
## Problem
As I commented #45, I think things are little unclear whether layers/functionals take waveforms or {real or complex} spectrograms. Magnitude downmix, for example, can (ideally) work for all three of them. Even if it doesn't, the name shouldn't be specifying what kind of input to take. Still, it'd be nicer if it can be specified intuitively. 

## Our current status

We haven't decided on it. There are `x`, `spect`, `signals`, etc. 

## Suggestion

I suggest to use different argument names for them -- instead of `tensor` or `x`, something like, e.g., `foo(waveforms, **kwargs)`, `foo(real_grams)` `foo(complex_spects)`. Librosa is a good example, it takes `S` as spectrogram input and `x` (or `y`?) for waveforms.
More specifically, I'm up for `waveforms` for, yes, a batch of waveforms. I think **`waveforms`** say everything about what it would be. On the batches of real or complex 2-d representations, I am not totally sure but..
- don't wanna dub `real_` to `r_`, or `complex` to `c_`. It adds another ambiguity which people would need to look up until they get used to.
- not a big fan of `M` (magnitude), `C` (complex), or `P` (phase) -- for the same reason above. + not very pep8 to use the capital letter (it's too matlab-y)
- hopefully, the name should sound like including all the `-grams` (ones with magnitude) + STFT. AFAIK the ~'gram'~(EDIT: `spec-`) implies it's a sort of visualization, hence it doesn't work for STFT. But I'm not sure how strong the feeling is. If it sounds ok for native speakers, I'm fine with `real_grams` and `complex_grams`. 

So, in short,

- +1 for `waveforms`
- +0.3 for `real_grams` and `complex_grams`
Any opinion? 
----------
[EDIT 1]
- +1 for `mag_specgrams`, `complex_specgrams`
- +0.7 for `mag_specs`, `complex_specs`
  - Because actually 'spec' is the informative part of the word. I feel like `spec` is also quite a common var name. 
  - `real` could be misleading since it could mean the real part of a complex number. 
  - `_specgrams` > `_specs`, just for clarity.

**Comment by ksanjeevan** [comment]
>We haven't decided on it. There are x, spect, signals, etc.

Yeah I was going off the discussion [here](https://github.com/keunwoochoi/torchaudio-contrib/issues/36#issuecomment-481144944) from #36. 

Looks like we'd be going for `signal -> waveform` and `spect -> _grams`. I definetley like adding the `real_`/`complex_`, even if we keep the `spect` I think it's better than `stft` for complex (which is just confusing as a param name) and `spect` for real.

> - don't wanna dub real_ to r_, or complex to c_. It adds another ambiguity which people would need to look up until they get used to.
> - not a big fan of M (magnitude), C (complex), or P (phase) -- for the same reason above. + not very pep8 to use the capital letter (it's too matlab-y)

:+1:

**Comment by faroit** [comment]
> complex_specgrams

there is a bunch of people in DSP that really don't like calling complex stfts a spectrogram. This also goes in line with many textbooks and wikipedia. I would really suggest to keep this accurate (even though I agree with you that the parameter naming doesn't read well).

**Comment by faroit** [comment]
another option could be using `image`, especially since we fully want to support multichannel. Such as `real_image`, `complex_image`.

**Comment by keunwoochoi** [comment]
@faroit Yeah, couldn't find a perfect name. But would `images` be less provoking than `specgrams`? :P I think the most important thing here is to **correctly imply**.
> I would really suggest to keep this accurate

I wish I could! But for an API, I'd like it to be specific as well as accurate (yeah, I'm not thinking like a dsp person here).

**Comment by f0k** [comment]
I think this should just be part of the discussion in #36, not a separate issue. Shall we close this and continue over there?

**Comment by keunwoochoi** [comment]
You're right, this is what we were talking about with `spect`. I'm closing it.
